### PR TITLE
Move About pgAdmin 4 menu option to top of application menu for macOS. #7289

### PR DESCRIPTION
**Pull request by RohitBhati8269**
Detail Info:-
1. https://github.com/pgadmin-org/pgadmin4/issues/7289
2. https://github.com/pgadmin-org/pgadmin4/issues/7988

### CODE REVIEW

File: runtime/src/js/menu.js [hunk 1]
```diff
@@ -41,6 +41,25 @@ function buildMenu(pgadminMenus, pgAdminMainScreen, callbacks) {
   });
 
   let menuFile = pgadminMenus.shift();
+
+  if (isMac) {
+    // Add About pgAdmin 4 on top of menu.
+    const aboutIndex = menuFile.submenu.findIndex((item) => item.label === 'About pgAdmin 4');
```
Comment (adityatoshniwal) [inline runtime/src/js/menu.js:47]: `About pgAdmin 4` can be any language not just english.

File: runtime/src/js/menu.js [hunk 2]
```diff
@@ -41,6 +41,25 @@ function buildMenu(pgadminMenus, pgAdminMainScreen, callbacks) {
...
+    if (aboutIndex !== -1) {
+      menuFile.submenu.splice(0, 0, menuFile.submenu.splice(aboutIndex, 1)[0]);
+      menuFile.submenu.splice(1, 0, { type: 'separator' });
+    }
+    // Remove About pgAdmin 4 from help menu.
+    const helpMenuIndex = pgadminMenus.findIndex((menu) => menu.label === 'Help');
```
Comment (adityatoshniwal) [inline runtime/src/js/menu.js:53]: `Help` can be in any language not just english.

File: runtime/src/js/menu.js [hunk 3]
```diff
@@ -41,6 +41,25 @@ function buildMenu(pgadminMenus, pgAdminMainScreen, callbacks) {
   });
 
   let menuFile = pgadminMenus.shift();
+
+  if (isMac) {
+    // Add About pgAdmin 4 on top of menu.
+    const aboutIndex = menuFile.submenu.findIndex((item) => item.label === 'About pgAdmin 4');
+    if (aboutIndex !== -1) {
+      menuFile.submenu.splice(0, 0, menuFile.submenu.splice(aboutIndex, 1)[0]);
```
Comment (adityatoshniwal) [inline runtime/src/js/menu.js:49]: You can directly add this to `template.push`. No need add so many splice.

File: runtime/src/js/menu.js [hunk 4]
```diff
@@ -41,6 +41,25 @@ function buildMenu(pgadminMenus, pgAdminMainScreen, callbacks) {
...
+      menuFile.submenu.splice(1, 0, { type: 'separator' });
+    }
+    // Remove About pgAdmin 4 from help menu.
+    const helpMenuIndex = pgadminMenus.findIndex((menu) => menu.label === 'Help');
+    if (helpMenuIndex !== -1) {
+      const helpMenu = pgadminMenus[helpMenuIndex];
+      const aboutIndexInHelpMenu = helpMenu.submenu.findIndex((item) => item.label === 'About pgAdmin 4');
```
Comment (adityatoshniwal) [inline runtime/src/js/menu.js:56]: You can use filter and assign. no need to splice.

File: web/pgadmin/settings/__init__.py [hunk 5]
```diff
@@ -38,6 +42,14 @@ def get_own_menuitems(self):
                     module="pgAdmin.Settings",
                     callback='show',
                     label=gettext('Reset Layout')
+                ),
+                MenuItem(
```
Comment (adityatoshniwal) [inline web/pgadmin/settings/__init__.py:46]: what is this change?

**Review by adityatoshniwal — CHANGES_REQUESTED** [review]

File: web/pgadmin/settings/__init__.py (hunk 5 above)
Comment (RohitBhati8269) [reply web/pgadmin/settings/__init__.py]: Previously, I was unable to fetch the **About pagamin 4** option until I added it, but that's my misunderstanding now I have fixed it.

**Pull request merged**